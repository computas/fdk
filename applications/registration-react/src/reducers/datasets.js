import {
  DATASETS_REQUEST,
  DATASETS_SUCCESS,
  DATASETS_FAILURE
} from '../constants/ActionTypes';

export default function datasets(
  state = { isFetchingDatasets: false, datasetItems: null },
  action
) {
  switch (action.type) {
    case DATASETS_REQUEST:
      return {
        ...state,
        isFetchingDatasets: true
      };
    case DATASETS_SUCCESS:
      return {
        ...state,
        isFetchingDatasets: false,
        datasetItems: action.response.data
      };
    case DATASETS_FAILURE:
      return {
        ...state,
        isFetchingDatasets: false,
        datasetItems: null
      };
    default:
      return state;
  }
}
