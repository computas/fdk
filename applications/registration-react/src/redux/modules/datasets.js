import _ from 'lodash';
import {
  DATASETS_REQUEST,
  DATASETS_SUCCESS,
  DATASETS_FAILURE
} from '../../constants/ActionTypes';
import { fetchActions } from '../fetchActions';

function shouldFetch(metaState) {
  const threshold = 60 * 1000; // seconds
  return (
    !metaState ||
    (!metaState.isFetching &&
      (metaState.lastFetch || 0) < Date.now() - threshold)
  );
}

export function fetchDatasetsIfNeeded(catalogId) {
  return (dispatch, getState) =>
    shouldFetch(_.get(getState(), ['datasets', catalogId, 'meta'])) &&
    dispatch(
      fetchActions(`/catalogs/${catalogId}/datasets?size=100`, [
        { type: DATASETS_REQUEST, meta: { catalogId } },
        { type: DATASETS_SUCCESS, meta: { catalogId } },
        { type: DATASETS_FAILURE, meta: { catalogId } }
      ])
    );
}

const initialState = {};

export default function datasets(state = initialState, action) {
  switch (action.type) {
    case DATASETS_REQUEST:
      return {
        ...state,
        [action.meta.catalogId]: {
          meta: {
            isFetching: true,
            lastFetch: null
          }
        }
      };
    case DATASETS_SUCCESS:
      return {
        ...state,
        [action.meta.catalogId]: {
          items: _.get(action.payload, ['_embedded', 'datasets']),
          meta: {
            isFetching: false,
            lastFetch: Date.now()
          }
        }
      };
    case DATASETS_FAILURE:
      return {
        ...state,
        [action.meta.catalogId]: {
          meta: {
            isFetching: false,
            lastFetch: null
          }
        }
      };
    default:
      return state;
  }
}

export const getDatasetItemsByCatalogId = (datasets, catalogId) =>
  _.get(datasets, [catalogId, 'items']);
