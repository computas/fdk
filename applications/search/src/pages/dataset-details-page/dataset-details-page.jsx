import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import DocumentMeta from 'react-document-meta';

import { DatasetDescription } from './dataset-description/dataset-description.component';
import { DatasetKeyInfo } from './dataset-key-info/dataset-key-info.component';
import { DatasetDistribution } from './dataset-distribution/dataset-distribution.component';
import { DatasetInfo } from './dataset-info/dataset-info.component';
import { DatasetQuality } from './dataset-quality/dataset-quality.component';
import { DatasetBegrep } from './dataset-begrep/dataset-begrep.component';
import { DatasetLandingPage } from './dataset-landing-page/dataset-landing-page.component';
import { DatasetContactInfo } from './dataset-contact-info/dataset-contact-info.component';
import localization from '../../lib/localization';
import { getTranslateText } from '../../lib/translateText';

export function DatasetDetailsPage(props) {
  function _renderDatasetDescription() {
    const { datasetItem } = props;
    return (
      <DatasetDescription
        title={getTranslateText(datasetItem.title)}
        description={getTranslateText(datasetItem.description)}
        descriptionFormatted={getTranslateText(
          datasetItem.descriptionFormatted
        )}
        objective={getTranslateText(datasetItem.objective)}
        publisher={datasetItem.publisher}
        themes={datasetItem.theme}
        harvest={datasetItem.harvest}
        provenance={datasetItem.provenance}
      />
    );
  }

  function _renderDistribution() {
    const { distribution, accessRights } = props.datasetItem;
    const { openLicenseItems, distributionTypeItems } = props;
    if (!distribution) {
      return null;
    }
    return distribution.map(distribution => (
      <DatasetDistribution
        key={encodeURIComponent(distribution.uri)}
        title={localization.dataset.distribution.title}
        description={getTranslateText(distribution.description)}
        accessUrl={distribution.accessURL}
        format={distribution.format}
        code={accessRights ? accessRights.code : null}
        license={distribution.license}
        conformsTo={distribution.conformsTo}
        page={distribution.page}
        type={distribution.type}
        openLicenseItems={openLicenseItems}
        distributionTypeItems={distributionTypeItems}
      />
    ));
  }

  function _renderSample() {
    const { sample } = props.datasetItem;
    if (!sample) {
      return null;
    }
    return sample.map(sample => (
      <DatasetDistribution
        key={encodeURIComponent(sample.uri)}
        title={localization.dataset.sample}
        description={getTranslateText(sample.description)}
        accessUrl={sample.accessURL}
        format={sample.format}
        code="SAMPLE"
      />
    ));
  }

  function _renderKeyInfo() {
    const { datasetItem } = props;
    return (
      <DatasetKeyInfo
        accessRights={datasetItem.accessRights}
        legalBasisForRestriction={datasetItem.legalBasisForRestriction}
        legalBasisForProcessing={datasetItem.legalBasisForProcessing}
        legalBasisForAccess={datasetItem.legalBasisForAccess}
        type={datasetItem.type}
        conformsTo={datasetItem.conformsTo}
        informationModel={datasetItem.informationModel}
      />
    );
  }

  function _renderDatasetInfo() {
    const {
      issued,
      accrualPeriodicity,
      modified,
      provenance,
      hasCurrentnessAnnotation,
      spatial,
      temporal,
      language,
      isPartOf,
      references
    } = props.datasetItem;

    return (
      <DatasetInfo
        issued={issued || null}
        accrualPeriodicity={getTranslateText(
          accrualPeriodicity && accrualPeriodicity.prefLabel
        )}
        modified={modified}
        provenance={getTranslateText(provenance && provenance.prefLabel)}
        hasCurrentnessAnnotation={getTranslateText(
          hasCurrentnessAnnotation && hasCurrentnessAnnotation.hasBody
        )}
        spatial={spatial}
        temporal={temporal}
        language={language}
        isPartOf={isPartOf}
        references={references}
      />
    );
  }

  function _renderQuality() {
    const {
      hasRelevanceAnnotation,
      hasCompletenessAnnotation,
      hasAccuracyAnnotation,
      hasAvailabilityAnnotation
    } = props.datasetItem;
    if (
      hasRelevanceAnnotation ||
      hasCompletenessAnnotation ||
      hasAccuracyAnnotation ||
      hasAvailabilityAnnotation
    ) {
      return (
        <DatasetQuality
          relevanceAnnotation={getTranslateText(
            hasRelevanceAnnotation && hasRelevanceAnnotation.hasBody
          )}
          completenessAnnotation={getTranslateText(
            hasCompletenessAnnotation && hasCompletenessAnnotation.hasBody
          )}
          accuracyAnnotation={getTranslateText(
            hasAccuracyAnnotation && hasAccuracyAnnotation.hasBody
          )}
          availabilityAnnotations={getTranslateText(
            hasAvailabilityAnnotation && hasAvailabilityAnnotation.hasBody
          )}
        />
      );
    }
    return null;
  }

  function _renderLandingPageAndContactInfo() {
    const { contactPoint, landingPage } = props.datasetItem;
    if (!(contactPoint || landingPage)) {
      return null;
    }
    const contactPoints = items =>
      items.map((item, index) => (
        <DatasetContactInfo
          key={index}
          landingPage={landingPage}
          contactPoint={item}
        />
      ));

    const landingPages = items =>
      items.map(item => (
        <DatasetLandingPage
          key={`dataset-contactpoint-${item.id}`}
          landingPage={item}
        />
      ));

    return (
      <section className="mt-5">
        {landingPage && landingPages(landingPage)}
        {contactPoint && contactPoints(contactPoint)}
      </section>
    );
  }

  function _renderBegrep() {
    const { keyword, subject } = props.datasetItem;
    if (keyword || subject) {
      return <DatasetBegrep keyword={keyword} subject={subject} />;
    }
    return null;
  }

  props.fetchDistributionTypeIfNeeded();

  const { datasetItem } = props;
  if (!datasetItem) {
    return null;
  }

  const meta = {
    title: getTranslateText(_.get(datasetItem, 'title')),
    description: getTranslateText(_.get(datasetItem, 'description'))
  };
  return (
    <div className="container">
      <div className="row">
        <div className="col-lg-8 offset-lg-2" id="content" role="main">
          <article>
            <DocumentMeta {...meta} />
            {_renderDatasetDescription()}
            {_renderKeyInfo()}
            {_renderDistribution()}
            {_renderSample()}
            {_renderDatasetInfo()}
            {_renderQuality()}
            {_renderBegrep()}
            {_renderLandingPageAndContactInfo()}
          </article>
        </div>
      </div>
    </div>
  );
}

DatasetDetailsPage.defaultProps = {
  datasetItem: null,
  distributionTypeItems: null,
  fetchDistributionTypeIfNeeded: () => {}
};

DatasetDetailsPage.propTypes = {
  datasetItem: PropTypes.object,
  distributionTypeItems: PropTypes.array,
  fetchDistributionTypeIfNeeded: PropTypes.func
};
