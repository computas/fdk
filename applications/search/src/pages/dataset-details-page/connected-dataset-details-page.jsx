import { connect } from 'react-redux';
import { fetchDistributionTypeIfNeededAction } from '../../redux/modules/distributionType';
import { ResolvedDatasetDetailsPage } from './resolved-dataset-details-page';

const mapStateToProps = ({ distributionTypes }) => {
  const { distributionTypeItems } = distributionTypes || {
    distributionTypeItems: null
  };

  return {
    distributionTypeItems
  };
};

const mapDispatchToProps = dispatch => ({
  fetchDistributionTypeIfNeeded: () =>
    dispatch(fetchDistributionTypeIfNeededAction())
});

export const ConnectedDatasetDetailsPage = connect(
  mapStateToProps,
  mapDispatchToProps
)(ResolvedDatasetDetailsPage);
