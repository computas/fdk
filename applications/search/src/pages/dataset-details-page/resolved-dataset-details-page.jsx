import _ from 'lodash';
import { resolve } from 'react-resolver';
import { getDataset } from '../../api/get-dataset';
import { DatasetDetailsPage } from './dataset-details-page';

const memoizedGetDataset = _.memoize(getDataset);

const mapProps = {
  datasetItem: props => memoizedGetDataset(props.match.params.id)
};

export const ResolvedDatasetDetailsPage = resolve(mapProps)(DatasetDetailsPage);
