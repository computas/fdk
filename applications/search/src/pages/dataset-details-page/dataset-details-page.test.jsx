import React from 'react';
import { shallow } from 'enzyme';
import { DatasetDetailsPage } from './dataset-details-page';
import datasets from '../../../test/fixtures/datasets';

let wrapper;

beforeEach(() => {
  wrapper = shallow(<DatasetDetailsPage datasetItem={datasets[0]._source} />);
});

test('should render DatasetDetailsPage correctly with no datasetItem', () => {
  const wrap = shallow(<DatasetDetailsPage />);
  expect(wrap).toMatchSnapshot();
});

test('should render DatasetDetailsPage correctly with datasetItem', () => {
  expect(wrapper).toMatchSnapshot();
});
