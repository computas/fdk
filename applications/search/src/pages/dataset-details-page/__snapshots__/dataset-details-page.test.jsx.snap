// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render DatasetDetailsPage correctly with datasetItem 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-lg-8 offset-lg-2"
      id="content"
      role="main"
    >
      <article>
        <SideEffect(DocumentMeta)
          description="Datasettet avgrenser område for virkeområdet til lov 6. juni 2009 nr. 35 om naturområder i Oslo og nærliggende kommuner (markaloven) som trådte i kraft 1. september 2009. Markalovens virkeområde er fastsatt i forskrift 4. september 2015 nr. 1032 om justering av markagrensen fastlegger markalovens geografiske virkeområde med tilhørende kart."
          title="AAMarkagrensen Oslo Kommune og nærliggende kommuner"
        />
        <DatasetDescription
          description="Datasettet avgrenser område for virkeområdet til lov 6. juni 2009 nr. 35 om naturområder i Oslo og nærliggende kommuner (markaloven) som trådte i kraft 1. september 2009. Markalovens virkeområde er fastsatt i forskrift 4. september 2015 nr. 1032 om justering av markagrensen fastlegger markalovens geografiske virkeområde med tilhørende kart."
          descriptionFormatted={null}
          harvest={null}
          objective="Datasettes formål er nullam quis rius eget urna mollis ornare vel eu leo. Praesent commodo cursus magna, vel scelerisque nisl consectetur et. Cras mattis consectetur purus sit amet fermentum. Vestibulum id ligula porta felis euismod semper."
          provenance={
            Object {
              "code": "vedtak",
              "prefLabel": Object {
                "nb": "Vedtak",
              },
              "uri": "http://data.brreg.no/datakatalog/provenance/vedtak",
            }
          }
          publisher={
            Object {
              "id": "987654321",
              "name": "TESTETATEN",
              "uri": "http://data.brreg.no/enhetsregisteret/enhet/987654321",
            }
          }
          themes={
            Array [
              Object {
                "title": Object {
                  "nb": "Forvaltning og offentlig støtte",
                },
                "uri": "http://publications.europa.eu/resource/authority/data-theme/GOVE",
              },
              Object {
                "title": Object {
                  "nb": "Miljø",
                },
                "uri": "http://publications.europa.eu/resource/authority/data-theme/ENVI",
              },
            ]
          }
          title="AAMarkagrensen Oslo Kommune og nærliggende kommuner"
        />
        <DatasetKeyInfo
          accessRights={
            Object {
              "code": "NON_PUBLIC",
              "prefLabel": Object {
                "nb": "Begrenset",
              },
              "uri": "http://publications.europa.eu/resource/authority/access-right/NON_PUBLIC",
            }
          }
          conformsTo={
            Array [
              Object {
                "extraType": "dct:Standard",
                "prefLabel": Object {
                  "nb": "SOSI",
                },
                "uri": "https://www.kartverket.no/geodataarbeid/standarder/sosi/",
              },
            ]
          }
          informationModel={
            Array [
              Object {
                "prefLabel": Object {
                  "nb": "SKOS",
                },
                "uri": "https://www.w3.org/2004/02/skos/",
              },
            ]
          }
          legalBasisForAccess={
            Array [
              Object {
                "prefLabel": Object {
                  "nb": "Den siste loven med den lengste tittelen § 10",
                },
                "uri": "http://lovdata/paragraph/10",
              },
            ]
          }
          legalBasisForProcessing={
            Array [
              Object {
                "prefLabel": Object {
                  "nb": "Den andre loven med lenger tittel § 2",
                },
                "uri": "http://lovdata/paragraph/2",
              },
            ]
          }
          legalBasisForRestriction={
            Array [
              Object {
                "prefLabel": Object {
                  "nb": "Lov om arkiv [arkivlova]",
                },
                "uri": "https://lovdata.no/dokument/NL/lov/1992-12-04-126",
              },
              Object {
                "prefLabel": Object {
                  "nb": "Den spesifikke loven § 20",
                },
                "uri": "http://lovdata/paragraph/20",
              },
              Object {
                "prefLabel": Object {
                  "nb": "Den mindre spesifikke loven § 26",
                },
                "uri": "http://lovdata/paragraph/26",
              },
            ]
          }
          type="Kodeliste"
        />
        <DatasetDistribution
          accessUrl={
            Array [
              "http://www.detteerenlenke.no/til-nedlasting",
              "http://www.detteerenannenlenke.no/til-en-annen-nedlasting
",
              "http://www.detteerentredjelenke.no/til-en-tredje-nedlasting",
            ]
          }
          code="NON_PUBLIC"
          conformsTo={null}
          description="Dette er beskrivelsen av distribusjonen. Vivamus sagittis lacus vel augue laoreet rutrum faucibus dolor auctor. Vestibulum id ligula porta felis euismod semper con desbit arum. Se dokumentasjon for denne distribusjonen."
          distributionTypeItems={null}
          format={null}
          key="http%3A%2F%2Fbrreg.no%2Fcatalogs%2F987654321%2Fdatasets%2F3645d5a8-6689-4fc1-8684-d3d6c2d47386%2Fdistributions%2F96d63bba-e624-4f39-ad26-26b8c6a68880"
          license={
            Object {
              "extraType": "dct:Standard",
              "prefLabel": Object {
                "nb": "SOSI",
              },
              "uri": "https://www.kartverket.no/geodataarbeid/standarder/sosi/",
            }
          }
          page={
            Array [
              Object {
                "prefLabel": Object {
                  "nb": "Dokumentasjon av distribusjonen",
                },
                "uri": "http://lenke/til/mer/info",
              },
            ]
          }
          title="Distribution"
          type={null}
        />
        <DatasetDistribution
          accessUrl={
            Array [
              "http://www.detteerenlenke.no/til-nedlasting",
              "http://www.detteerenannenlenke.no/til-en-annen-nedlasting
",
              "http://www.detteerentredjelenke.no/til-en-tredje-nedlasting",
            ]
          }
          code="NON_PUBLIC"
          conformsTo={null}
          description="Dette er beskrivelsen av distribusjonen. Vivamus sagittis lacus vel augue laoreet rutrum faucibus dolor auctor. Vestibulum id ligula porta felis euismod semper con desbit arum. Se dokumentasjon for denne distribusjonen."
          distributionTypeItems={null}
          format={null}
          key="http%3A%2F%2Fbrreg.no%2Fcatalogs%2F987654321%2Fdatasets%2F3645d5a8-6689-4fc1-8684-d3d6c2d47386%2Fdistributions%2F96d63bba-e624-4f39-ad26-26b8c6a68880"
          license={
            Object {
              "extraType": "dct:Standard",
              "uri": "http://data.norge.no/nlod",
            }
          }
          page={
            Array [
              Object {
                "prefLabel": Object {
                  "nb": "Dokumentasjon av distribusjonen",
                },
                "uri": "http://lenke/til/mer/info",
              },
            ]
          }
          title="Distribution"
          type={null}
        />
        <DatasetDistribution
          accessUrl={
            Array [
              "http://www.detteerenlenke.no/til-nedlasting",
            ]
          }
          code="SAMPLE"
          conformsTo={null}
          description="Dette er beskrivelsen av eksempeldataene. Vivamus sagittis lacus vel augue laoreet rutrum faucibus dolor auctor."
          distributionTypeItems={null}
          format={
            Array [
              "application/rdf+xml",
            ]
          }
          key="http%3A%2F%2Fbrreg.no%2Fcatalogs%2F987654321%2Fdatasets%2F3645d5a8-6689-4fc1-8684-d3d6c2d47386%2Fdistributions%2F95a6b5e5-753c-4bc3-a421-2671c8f6a7bb"
          license={null}
          page={null}
          title="Sample data"
          type={null}
        />
        <DatasetInfo
          accrualPeriodicity="årlig"
          hasCurrentnessAnnotation="Denne teksten sier noe om aktualiteten. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras mattis consectetur purus sit amet fermentum. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus."
          issued="2018-04-26T00:00:00+0200"
          language={
            Array [
              Object {
                "code": "NOR",
                "prefLabel": Object {
                  "nb": "Norsk",
                },
                "uri": "http://publications.europa.eu/resource/authority/language/NOR",
              },
            ]
          }
          modified="2018-04-26T00:00:00+0200"
          provenance="Vedtak"
          references={
            Array [
              Object {
                "referenceType": Object {
                  "code": "hasVersion",
                  "prefLabel": Object {
                    "nb": "Har versjon",
                  },
                  "uri": "http://www.w3.org/2002/07/hasVersion",
                },
                "source": Object {
                  "contactPoint": Array [],
                  "description": Object {},
                  "legalBasisForAccess": Array [],
                  "legalBasisForProcessing": Array [],
                  "legalBasisForRestriction": Array [],
                  "registrationStatus": "DRAFT",
                  "title": Object {
                    "nb": "The first one",
                  },
                  "uri": "http://referenced/dataset",
                },
              },
            ]
          }
          spatial={
            Array [
              Object {
                "prefLabel": Object {
                  "nb": "Asker",
                },
                "uri": "http://www.geonames.org/3162656/asker.html",
              },
              Object {
                "prefLabel": Object {
                  "nb": "Bærum",
                },
                "uri": "http://www.geonames.org/3162212/baerum.html",
              },
              Object {
                "prefLabel": Object {
                  "nb": "Hurum",
                },
                "uri": "http://www.geonames.org/3151404/hurum.html",
              },
              Object {
                "prefLabel": Object {
                  "nb": "Røyken",
                },
                "uri": "http://www.geonames.org/3141104/royken.html",
              },
            ]
          }
          temporal={Array []}
        />
        <DatasetQuality
          accuracyAnnotation="Denne teksten sier noe om nøyaktigheten. Cras mattis consectetur purus sit."
          availabilityAnnotations={null}
          completenessAnnotation="Denne teksten sier noe om komplettheten. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras mattis consectetur purus sit amet fermentum."
          relevanceAnnotation="Denne teksten sier noe om relevansen. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras mattis consectetur purus sit amet fermentum. Cum sociis natoque penatibus et magnis dis parturient montes."
        />
        <DatasetBegrep
          keyword={
            Array [
              Object {
                "nb": "Bestemmelse",
              },
              Object {
                "nb": "jord",
              },
              Object {
                "nb": "regulering",
              },
              Object {
                "nb": "statlig bestemmelse",
              },
            ]
          }
          subject={
            Array [
              Object {
                "definition": Object {
                  "no": "alt som er registrert med et organisasjonsnummer ",
                },
                "note": Object {
                  "no": "Alle hovedenheter, underenheter og organisasjonsledd som er identifisert med et organisasjonsnummer.",
                },
                "prefLabel": Object {
                  "no": "enhet",
                },
                "source": "https://jira.brreg.no/browse/BEGREP-208",
                "uri": "https://data-david.github.io/Begrep/begrep/Enhet",
              },
            ]
          }
        />
        <section
          className="mt-5"
        >
          <DatasetLandingPage
            key="dataset-contactpoint-undefined"
            landingPage="http://testetaten.no/landingsside/nr1"
          />
          <DatasetContactInfo
            contactPoint={
              Object {
                "email": "digitalisering@kartverket.no",
                "hasTelephone": "22306022",
                "hasURL": "http://testetaten.no/url",
                "id": "3c50088c-489f-4168-95e6-e6f3cd0c67c5",
                "organizationUnit": "Avdeling for digitalisering",
                "uri": "http://brreg.no/catalogs/987654321/contacts/null",
              }
            }
            key="0"
            landingPage={
              Array [
                "http://testetaten.no/landingsside/nr1",
              ]
            }
          />
        </section>
      </article>
    </div>
  </div>
</div>
`;

exports[`should render DatasetDetailsPage correctly with no datasetItem 1`] = `""`;
